modules/common/src/main/ts/api/Reporter.ts

```diff
@@ -30,111 +22,72 @@ const processQUnit = (html: string): string =>
-  const actual = stringify(e.actual);
-  const expected = stringify(e.expected);
-  const message = `Assertion error: ${e.message ? escape(e.message) : ''}`;
-  if (e.showDiff !== false) {
-    const dh = diff(actual, expected);
-    return `${message}
-Expected:
-${escape(expected)}
-Actual:
-${escape(actual)}
-Diff:
-${dh}`;
+const pprintBasicError = (e: BasicErrorData, escape: (value: string) => string, diff: DifferFn): string => {
+  const message = escape(e.message);
+  const diffMessage = pprintDiff(e, escape, diff);
+  if (diffMessage.length > 0) {
+    return `${message}\n${diffMessage}`;
   } else {
     return message;
   }
 };
 
-const assertionErrorHtml = (e: AssertionError): string => assertionError(e, htmlentities, Differ.diffPrettyHtml);
-const assertionErrorText = (e: AssertionError): string => assertionError(e, identity, Differ.diffPrettyText);
-
-const mkHtml = (e: TestError): string => {
-  if (TestError.isHTMLDiffError(e)) {
-    return htmlDiffAssertionErrorHtml(e);
-  } else if (TestError.isPprintAssertionError(e)) {
-    return pprintAssertionErrorHtml(e);
-  } else if (TestError.isAssertionError(e)) {
-    return assertionErrorHtml(e);
-  } else if (e.name && e.message) {
-    return htmlentities(e.name + ': ' + e.message);
-  } else if (e.toString !== undefined) {
-    return htmlentities(String(e));
-  } else {
-    return htmlentities(JSON.stringify(e));
-  }
+const pprintError = (e: ErrorData, escape: (value: string) => string, diff: DifferFn): string => {
+  const message = pprintBasicError(e, escape, diff);
+  const extras = escape(pprintExtra(e));
+  return `${message}${extras}`;
 };
 
-const mkText = (e: TestError): string => {
-  if (TestError.isHTMLDiffError(e)) {
-    return htmlDiffAssertionErrorText(e);
-  } else if (TestError.isPprintAssertionError(e)) {
-    return pprintAssertionErrorText(e);
-  } else if (TestError.isAssertionError(e)) {
-    return assertionErrorText(e);
-  } else if (e.name && e.message) {
-    return (e.name + ': ' + e.message);
-  } else if (e.toString !== undefined) {
-    return String(e);
-  } else {
-    return JSON.stringify(e);
-  }
+export const pprintAssertionError = (e: PprintAssertionError): string => {
+  const err = ErrorExtractor.getBasicErrorData(e);
+  return pprintBasicError(err, identity, Differ.diffPrettyText);
 };
 
+export const data = (err: LoggedError): ErrorData =>
+  ErrorExtractor.getErrorData(err);
+
+export const dataHtml = (err: ErrorData): string =>
+  pprintError(err, htmlentities, (actual, expected, comparison) => {
+    return comparison !== undefined ? processQUnit(comparison) : Differ.diffPrettyHtml(actual, expected);
+  });
+
+export const dataText = (err: ErrorData): string =>
+  pprintError(err, identity, (actual, expected, comparison) => {
+    // TODO: get rid of the <ins> and <del> in the comparison. Probably need to change the code that throws HtmlAssertionError.
```

Comment:
Does this need a Jira?

Reply:
Moved code and was like this previously. So out of scope, however in saying that IMO it's not important as we are moving away from all the old assertion libraries and this will eventually become irrelvant.